cid/helpers/cur_proxy.py

```diff
@@ -0,0 +1,452 @@
+        self.fields_to_expose = list(set(default_columns[self.target_cur_version] + (fields_to_expose or []) ))
+        self.athena = self.cur.athena
+        self.name = 'cur1_proxy'
+        self.exposed_fields = []
+
+    def read_from_athena(self):
+        view_name = self.name
+        _mapping = {
+            'timestamp(3)': 'datetime',
+            'varchar': 'string',
+        }
+        exposed_fields_and_types = dict(self.athena.query(f'''
+            SELECT column_name, data_type
+            FROM information_schema.columns
+            WHERE table_schema = '{self.athena.DatabaseName}'
+            AND table_name = '{view_name}';
+        '''))
+        self.exposed_fields = list(exposed_fields_and_types.keys())
+        logger.debug(self.exposed_fields)
+
+    def get_sql_expression(self, field, field_type):
+        if self.current_cur_version.startswith('2') and self.target_cur_version.startswith('2'):
+            return field
+        if self.current_cur_version.startswith('1') and self.target_cur_version.startswith('1'):
+            return field
+        if self.current_cur_version.startswith('1') and self.target_cur_version.startswith('2'):
+            if field_type.startswith('map'):
+                map_mapping = {}
+                for cur1_field in self.cur.fields:
+                    if cur1_field.startswith(field + '_'):
+                        map_mapping[cur1_field[len(field + '_'):]] = cur1_field
+                if not map_mapping:
+                    return 'cast(NULL AS MAP<VARCHAR, VARCHAR>)'
+                return f'''
+                    MAP(
+                        ARRAY[{', '.join(["'" + key + "'" for key in map_mapping.keys()])}],
+                        ARRAY[{', '.join([cur1_field for cur1_field in map_mapping.values()])}]
+                    )
+                '''
+            cur1to2mapping = {value: key for key, value in cur2to1mapping.items()}
+            if field in cur1to2mapping:
+                return f'{cur1to2mapping.ge(field, field)}'
+            else:
+                raise NotImplementedError(f'WARNING: {field} has not known equivalent')
+
+        if self.current_cur_version.startswith('2') and self.target_cur_version.startswith('1'):
+            if field.startswith('resource_tags_'):
+                return f"resource_tags['{field[len('resource_tags_'):]}']"
+            if field.startswith('cost_category_'):
+                return f"cost_category['{field[len('cost_category_'):]}']"
+            return cur2to1mapping.get(field, field)
+
+    def create_or_update_view(self):
+        self.read_from_athena()
+        all_fields  = list(set(self.exposed_fields + self.fields_to_expose))
+        lines = []
+        print('all_fields',  all_fields)
+        for field in all_fields:
+            field_type = self.cur.get_type_of_column(field)
+            mapped_expression = self.get_sql_expression(field, field_type)
+            requirement = mapped_expression.split('[')[0]
```

Comment:
What is purpose of this?

Reply:
For a field product['something'] the requirement would be product